### PR TITLE
Add links to relevant pages from config reference sections

### PR DESCRIPTION
**Pull request by rosieyohannan**
In this PR I have:

* added links to relevant conceptual and how-to docs where we have them
* Moved some content out of the config reference to the concepts docs to keep the config reference more concise - there is much more to do here - if people don't like what I've done here totally happy to revert and rethink.
* Removed no_toc tags so everything appears in the right hand nav, and changed some example sections just use **example** rather than actual headings.

### CODE REVIEW

File: jekyll/_cci2/configuration-reference.md [hunk 1]
```diff
@@ -230,49 +229,33 @@ See [Parameter Syntax]({{ site.baseurl }}/2.0/reusing-config/#parameter-syntax)
...
+entrypoint | N | String or List | The command used as executable when launching the container. `entrypoint` overrides the image's [`ENTRYPOINT`](https://docs.docker.com/engine/reference/builder/#entrypoint).
+command | N | String or List | The command used as pid 1 (or args for entrypoint) when launching the container. `command` overrides the image's `COMMAND`. It will be used as arguments to the image `ENTRYPOINT` if it has one, or as the executable if the image has no `ENTRYPOINT`.
 user | N | String | Which user to run commands as within the Docker container
-environment | N | Map | A map of environment variable names and values
+environment | N | Map | A map of environment variable names and values. The `environment` settings apply to the entrypoint/command run by the docker container, not the job steps.
 auth | N | Map | Authentication for registries using standard `docker login` credentials
 aws_auth | N | Map | Authentication for AWS Elastic Container Registry (ECR)
 {: class="table table-striped"}
 
-The first `image` listed under a job defines the job's own primary container image where all steps will run.
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:253]: I've just moved this info into the table to make the whole thing a bit more succinct.

File: jekyll/_cci2/configuration-reference.md [hunk 2]
```diff
@@ -323,30 +306,14 @@ jobs:
           aws_secret_access_key: $ECR_AWS_SECRET_ACCESS_KEY  # or project UI envar reference
 ```
 
-It is possible to reuse [declared commands]({{ site.baseurl }}/2.0/reusing-config/) in a job when using version 2.1. The following example invokes the `sayhello` command.
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:326]: This example was not relevant to this section.

File: jekyll/_cci2/configuration-reference.md [hunk 3]
```diff
@@ -522,7 +448,20 @@ We implement soft concurrency limits for each resource class to ensure our syste
 
 **Note:** If you do not specify a resource class, CircleCI will use a default value that is subject to change.  It is best practice to specify a resource class as opposed to relying on a default.
 
-**For self-hosted installations of CircleCI Server contact your system administrator for a list of available resource classes**. See Server Administration documents for further information: [Nomad Client System Requirements]({{ site.baseurl }}/2.0/server-ports/#nomad-clients) and [Server Resource Classes]({{ site.baseurl }}/2.0/customizations/#resource-classes).
+**For self-hosted installations of CircleCI Server contact your system administrator for a list of available resource classes**.
+
+##### Self-hosted runner
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:453]:
```suggestion
##### Self-hosted runner
{: #self-hosted-runner }
```

File: jekyll/_cci2/configuration-reference.md [hunk 4]
```diff
@@ -1727,7 +1613,10 @@ A job may have a `type` of `approval` indicating it must be manually approved be
 
 ###### **`filters`**
 {: #jobfilters }
-Job Filters can have the key `branches` or `tags`. **Note** Workflows will ignore job-level branching. If you use job-level branching and later add workflows, you must remove the branching at the job level and instead declare it in the workflows section of your `config.yml`, as follows:
+
+Job Filters can have the key `branches` or `tags`.
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:1617]:
```suggestion
Job Filters can have the key `branches` or `tags`.
```

File: jekyll/_cci2/configuration-reference.md [hunk 5]
```diff
@@ -211,7 +210,7 @@ jobs:
 
 #### `parameters`
 {: #parameters }
-The `parameters` can be used when [calling that `job` in a `workflow`](#jobs-in-workflow).
+`parameters` can be used when [calling that `job` in a `workflow`](#jobs-in-workflow).
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:213]:
```suggestion
`parameters` can be used when [calling a `job` in a `workflow`](#jobs-in-workflow).
```

File: jekyll/_cci2/configuration-reference.md [hunk 6]
```diff
@@ -522,7 +448,21 @@ We implement soft concurrency limits for each resource class to ensure our syste
...
+Use the `resource_class` to configure a [self-hosted runner instance](https://circleci.com/docs/2.0/runner-overview/).
+
+For example:
+
+```yaml
+jobs:
+  job_name:
+    machine: true
+    resource_class: my-namespace/my-runner
+```
```
Comment (rosieyohannan) [inline jekyll/_cci2/configuration-reference.md:465]:
```suggestion
Use the `resource_class` key to configure a [self-hosted runner instance](https://circleci.com/docs/2.0/runner-overview/).

For example:

```yaml
jobs:
  job_name:
    machine: true
    resource_class: <my-namespace>/<my-runner>
```
```

File: jekyll/_cci2/configuration-reference.md [hunk 7]
```diff
@@ -182,13 +182,12 @@ resource_class | N | String | Amount of CPU and RAM allocated to each container
...
 
 #### `parallelism`
 {: #parallelism }
 
-If `parallelism` is set to N > 1, then N independent executors will be set up and each will run the steps of that job in parallel. This can help optimize your test steps; you can split your test suite, using the CircleCI CLI, across parallel containers so the job will complete in a shorter time. Certain parallelism-aware steps can opt out of the parallelism and only run on a single executor. Learn more about [parallel jobs]({{ site.baseurl }}/2.0/parallelism-faster-jobs/).
+If `parallelism` is set to N > 1, then N independent executors will be set up and each will run the steps of that job in parallel. This features is used to optimize your test steps. Split your test suite, using the CircleCI CLI, across parallel containers so the job will complete in a shorter time. Certain parallelism-aware steps can opt out of the parallelism and only run on a single executor. Learn more on the [Running Tests in Parallel]({{ site.baseurl }}/2.0/parallelism-faster-jobs/) page.
```
Comment (catherinepope) [inline jekyll/_cci2/configuration-reference.md:190]:
```suggestion
If `parallelism` is set to N > 1, then N independent executors will be set up and each will run the steps of that job in parallel. This feature is used to optimize your test steps. Split your test suite, using the CircleCI CLI, across parallel containers so the job will complete in a shorter time. Certain parallelism-aware steps can opt out of the parallelism and only run on a single executor. Learn more on the [Running Tests in Parallel]({{ site.baseurl }}/2.0/parallelism-faster-jobs/) page.
```

File: jekyll/_cci2/workspaces.md [hunk 8]
```diff
@@ -19,6 +19,14 @@ For example, a project with a `build` job that builds a `.jar` file and saves it
 
 Workspaces are additive-only data storage. Jobs can persist data to the workspace. When a workspace is used, data is archived and stored in an off-container store. With each addition to the workspace a new layer is created in the store. Downstream jobs can then attach the workspace to their container filesystem. Attaching the workspace downloads and unpacks each layer based on the ordering of the upstream jobs in the workflow.
 
+Some notes about workspaces:
+
+* Each workflow has a temporary workspace associated with it. The workspace can be used to pass along unique data built during a job to other jobs in the same workflow.
+* Jobs can add files into the workspace using the `persist_to_workspace` step and download the workspace content into their file system using the `attach_workspace` step.
+* The workspace is additive only, jobs may add files to the workspace but cannot delete files from the workspace. * Each job can only see content added to the workspace by the jobs that are upstream of it.
```
Comment (catherinepope) [inline jekyll/_cci2/workspaces.md:26]:
```suggestion
* The workspace is additive only, jobs may add files to the workspace but cannot delete files from the workspace. 
* Each job can only see content added to the workspace by the jobs that are upstream of it.
```

File: jekyll/_cci2/workspaces.md [hunk 9]
```diff
@@ -19,6 +19,14 @@ For example, a project with a `build` job that builds a `.jar` file and saves it
...
+* When attaching a workspace the "layer" from each upstream job is applied in the order the upstream jobs appear in the workflow graph. When two jobs run concurrently, the order in which their layers are applied is undefined. * If multiple concurrent jobs persist the same filename then attaching the workspace will error.
```
Comment (catherinepope) [inline jekyll/_cci2/workspaces.md:27]:
```suggestion
* When attaching a workspace the "layer" from each upstream job is applied in the order the upstream jobs appear in the workflow graph. When two jobs run concurrently, the order in which their layers are applied is undefined. 
* If multiple concurrent jobs persist the same filename then attaching the workspace will error.
```

**Review by catherinepope — APPROVED** [review]
👍

**Review by catherinepope — APPROVED** [review]
👍 Great stuff!

**Review by catherinepope — APPROVED** [review]
👍 Great stuff!

**Pull request merged**